src/metric/SampleMetricManager.py

```diff
@@ -327,7 +351,8 @@ def calc(
                 )
 
             # Check valid result dimensions
-            if result.shape[0] != len(ids):
+            # Not applicable when using filtering
+            if result.shape[0] != len(ids) and "filter_bit" not in parameters:
```

Comment:
why is "filter bit" here? Should not be intrinsic for the existence of a sample metric? As I understand it `SampleMetricFilter` is utilizing the SMM, not the other way around